lnwallet/channel.go

```diff
@@ -2835,10 +2835,9 @@ func (lc *LightningChannel) fetchParent(entry *paymentDescriptor,
 // was committed is updated. Keeping track of this inclusion height allows us to
 // later compact the log once the change is fully committed in both chains.
 func processAddEntry(htlc *paymentDescriptor, ourBalance,
-	theirBalance *lnwire.MilliSatoshi, _ lntypes.ChannelParty,
-	isIncoming bool) {
+	theirBalance *lnwire.MilliSatoshi, _, originator lntypes.ChannelParty) {
 
-	if isIncoming {
+	if originator == lntypes.Remote {
```

Comment:
I think `isIncoming` is easier to understand - when I see `isIncoming` I know it's an incoming HTLC, when I see the originator is the remote I have to think about it.

Reply:
Yeah that's fair, the issue with `isIncoming` is that it promotes a boolean argument which I find doesn't fit the domain modeling as well. We can move the LinkDirection type to the lntypes package and then have the functions

```golang
func (p ChannelParty) SendDirection() LinkDirection {...}
func (p ChannelParty) RecvDirection() LinkDirection {...}
func (d LinkDirection) Origin() ChannelParty {...}
func (d LinkDirection) Target() ChannelParty {...}
```